app/src/main/java/io/novafoundation/nova/app/root/presentation/deepLinks/handlers/DAppDeepLinkHandler.kt

```diff
@@ -27,12 +30,13 @@ class DAppDeepLinkHandler(
     }
 
     override suspend fun handleDeepLink(data: Uri) {
-        // TODO: check that user has accounts here
-        val url = data.getDappUrl() ?: return
-        val normalizedUrl = Urls.normalizeUrl(url)
+        if (!accountRepository.hasMetaAccounts()) return
+
+        val url = data.getDappUrl() ?: throw DAppHandlingException.UrlIsInvalid
+        val normalizedUrl = runCatching { Urls.normalizeUrl(url) }.getOrNull() ?: throw DAppHandlingException.UrlIsInvalid
 
         val dAppMetadata = dappRepository.syncAndGetDapp(normalizedUrl)
-        if (dAppMetadata == null) return // TODO: handle error
+        if (dAppMetadata == null) throw DAppHandlingException.DomainIsNotMatched(normalizedUrl)
```

Comment:
All of checks should go after `automaticInteractionGate.awaitInteractionAllowed()` to avoid showing dialog before pin is entered. Please check other places as well